server/src/main/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregator.java

```diff
@@ -40,22 +40,23 @@
 /**
  * An aggregator that computes approximate counts of unique values.
  */
-public class CardinalityAggregator extends NumericMetricsAggregator.SingleValue {
+public abstract class CardinalityAggregator extends NumericMetricsAggregator.SingleValue {
 
-    private final int precision;
+    protected final int precision;
+    protected ValuesSourceConfig valuesSourceConfig;
     private final ValuesSource valuesSource;
 
     // Expensive to initialize, so we only initialize it when we have an actual value source
     @Nullable
-    private HyperLogLogPlusPlus counts;
+    protected HyperLogLogPlusPlus counts;
 
     private Collector collector;
 
-    private int emptyCollectorsUsed;
-    private int numericCollectorsUsed;
-    private int ordinalsCollectorsUsed;
-    private int ordinalsCollectorsOverheadTooHigh;
-    private int stringHashingCollectorsUsed;
+    protected int emptyCollectorsUsed;
+    protected int numericCollectorsUsed;
+    protected int ordinalsCollectorsUsed;
+    protected int ordinalsCollectorsOverheadTooHigh;
+    protected int stringHashingCollectorsUsed;
```

Comment:
I think these might not make sense to have any more because you can use the name of the aggregator to figure out which collector it made - except for the one that does heuristics on ordinals. That one probably just needs the last three. I think.

Reply:
Yeah, I think it's safe to get rid of the numeric count.  I'll do that.